features/frameworks/greentea-client/source/greentea_metrics.cpp

```diff
@@ -86,22 +85,21 @@ MBED_UNUSED static void send_stack_info()
     }
 
     // Print info for all other threads
-    osThreadEnumId enum_id = _osThreadsEnumStart();
-    while (true) {
-        osThreadId thread_id = _osThreadEnumNext(enum_id);
-        if (NULL == thread_id) {
-            // End of enumeration
-            break;
-        }
-        enqeue_thread_info(thread_id);
+    uint32_t thread_n = osThreadGetCount();
+    osThreadId_t *threads = new osThreadId_t[thread_n];
+    thread_n = osThreadEnumerate(threads, thread_n);
```

Comment:
The call to osThreadEnumerate and for loop after needs to wrapped in osKernelLock/Unlock to prevent tasks from changing state or getting created or destroyed in the middle of processing. Same for below